### PR TITLE
Add test for pipeline restarting

### PR DESCRIPTION
**Pull request by banasraf**
Add a test case for recovering from pipeline failure. 

Signed-off-by: Rafal <Banas.Rafal97@gmail.com>

### CODE REVIEW

File: src/dali_executor/executor.test.cc [hunk 1]
```diff
@@ -86,4 +86,56 @@ TEST_CASE("Scaling Pipeline")
...
+    auto output = executor.Run(std::vector<IODescr<false>>({input}));
+    std::vector<float> output_buffer(output[0].shape.num_elements());
+    std::vector<IODescr<false>> output_vec(1);
+    auto& outdesc = output_vec[0];
+    outdesc.device = device_type_t::GPU;
+    outdesc.device_id = 0;
+    outdesc.buffer = make_span(
+        (char*)output_buffer.data(),
+        output_buffer.size() * sizeof(decltype(output_buffer)::size_type));
+    executor.PutOutputs(output_vec);
```
Comment (JanuszL) [inline src/dali_executor/executor.test.cc:114]: Can we validate the output anyhow?

Comment (banasraf) [reply src/dali_executor/executor.test.cc:114]: Hard to say. I could validate the shape but I'm not sure about contents as it's Resnet pipeline run on some random small image. The purpose of this test was rather to check for exceptions

Comment (JanuszL) [reply src/dali_executor/executor.test.cc:114]: If the image is the same, I guess we can expect rather quite accurate Top1 or at least Top5.

Comment (banasraf) [reply src/dali_executor/executor.test.cc:118]: The image was single (0, 0, 0) pixel, so now I verify the output values.

**Review by JanuszL — APPROVED** [review]

**Pull request merged**